plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2824,10 +2824,6 @@ private void cleanExtraOutputFiles(ConnectorSession session, String queryId, Loc
         try {
             LOG.debug("Deleting failed attempt files from %s for query %s", location, queryId);
             TrinoFileSystem fileSystem = fileSystemFactory.create(session);
-            if (!fileSystem.newInputFile(location).exists()) {
-                // directory may not exist if no files were actually written
-                return;
```

Comment:
is `fileSystem.listFiles(location)` called below guaranteed to succeed on a non-existent directory?

Reply:
From the Javadoc:

> If the location does not exist, an empty iterator is returned.